lib/validators/icons-validator.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Validates:
+ * - Whether icons have the correct format
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import { Validator } from './validator';
+import { ComponentsDefinition } from '../models';
+var PNG = require('pngjs').PNG;
+
+export class IconsValidator implements Validator {
+
+    private supportedFormats = ['.svg', '.png'];
+
+    constructor(
+        private definition: ComponentsDefinition
+    ) {
+    }
+
+    validate(
+        errorReporter: (errorMessage: string) => void,
+    ): boolean {
+        let valid = true;
+
+        Object.values(this.definition.components).forEach((component) => {
+            const ext = path.extname(component.icon).toLowerCase();
+            if (this.supportedFormats.indexOf(ext) === -1) {
+                errorReporter(`Icons are only supported in SVG or transparent PNG format`);
+                valid = false;
```

Comment:
Should we denied other formats at all, or just show a warning message that it may look bad in the end?

Reply:
We could start with this and wait for feedback from customers?

Reply:
From what I understood it was decided a while ago to explicitly deny other file formats. This was also OK'd by Tom etc. (It's also described as such in the jira issue.)